flocker/control/test/test_persistence.py

```diff
@@ -830,3 +834,125 @@ def test_can_create_latest_golden(self):
                 "an upgrade test if you are intentionally changing the "
                 "model." % (path.path,)
             )
+
+
+class GenerationHashTests(TestCase):
+    """
+    Tests for generation_hash.
+    """
+
+    @given(st.data())
+    def test_no_hash_collisions(self, data):
+        """
+        Hashes of different deployments do not have hash collisions.
```

Comment:
I struggle to understand this test. Advanced hypothesis by the looks of it! Please expand the docstring and add some comments.